tethys_cli/db_commands.py

```diff
@@ -0,0 +1,144 @@
+                           superuser_name='tethys_super', superuser_password='pass',
+                           portal_superuser_name='admin', portal_superuser_email='', portal_superuser_password='pass')
+
+
+def init_db_server(db_dir=None, **kwargs):
+    args = ['initdb', '-U', 'postgres', '-D', f'{db_dir}/data']
+    run_process(args)
+
+
+def start_db_server(db_dir=None, port=None, **kwargs):
+    args = ['pg_ctl', '-U', 'postgres', '-D', f'{db_dir}/data', '-l', f'{db_dir}/logfile', 'start', '-o', f'-p {port}']
+    run_process(args)
+
+
+def stop_db_server(db_dir=None, **kwargs):
+    args = ['pg_ctl', '-U', 'postgres', '-D', f'{db_dir}/data', 'stop']
+    run_process(args)
+
+
+def create_db_user(port=None, username=None, password=None, db_name=None, is_superuser=False, **kwargs):
+    db_name = db_name or username
+
+    if is_superuser:
+        command = f"CREATE USER {username} WITH CREATEDB NOCREATEROLE SUPERUSER PASSWORD '{password}';"
+    else:
+        command = f"CREATE USER {username} WITH NOCREATEDB NOCREATEROLE NOSUPERUSER PASSWORD '{password}';"
+
+    args = ['psql', '-U', 'postgres', '-p', f'{port}', '--command', command]
+    run_process(args)
+    if not is_superuser:
+        args = ['createdb', '-U', 'postgres', '-p', f'{port}', '-O', username, db_name, '-E', 'utf-8', '-T',
+                'template0']
+        run_process(args)
+
+
+def create_tethys_db(port=None, db_name=None, username=None, password=None,
+                     superuser_name=None, superuser_password=None, **kwargs):
+    create_db_user(port=port, username=username, password=password, db_name=db_name)
+    if superuser_name is not None and superuser_password is not None:
+        create_db_user(port=port, username=superuser_name, password=superuser_password, is_superuser=True)
+
+
+def sync_tethys_db(db_alias=None, **kwargs):
+    manage_path = get_manage_path(None)
+    db_alias = db_alias or 'default'
+    args = ['python', manage_path, 'makemigrations']
+    run_process(args)
+
+    args = ['python', manage_path, 'migrate', '--database', db_alias]
+    run_process(args)
+
+
+def create_portal_superuser(portal_superuser_name='admin', portal_superuser_email='', portal_superuser_password='pass',
+                            **kwargs):
+    load_apps()
+    from django.contrib.auth.models import User  # noqa: E402
+    User.objects.create_superuser(portal_superuser_name, portal_superuser_email, portal_superuser_password)
+
+
+def configure_tethys_db(**kwargs):
+    init_db_server(**kwargs)
+    start_db_server(**kwargs)
+    create_tethys_db(**kwargs)
+    sync_tethys_db(**kwargs)
+    create_portal_superuser(**kwargs)
+
+
+def process_args(args):
+    db_settings = settings.DATABASES[args.db_alias]
+    try:
+        db_dir = db_settings['DIR']
+    except KeyError:
+        raise RuntimeError('The tethys db command can only be used with local databases.')
```

Comment:
Why this limitation? Can this command not be used to setup a Tethys DB in a docker container? What would it take to support that?

Reply:
I've changed this so it only requires a local db for some of the commands `init`, `start`, `stop`, and the composite command `configure`. So, `create` which creates the Tethys DBs, `sync` and `createsuperuser` should all be able to be run with any postgresql database server (although I have not verified this).